File: VMMVExample1/app/src/main/java/com/example/vmmvexample1/MainActivity.java
package com.example.vmmvexample1;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;
import androidx.lifecycle.ViewModelProviders;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.view.View;
import android.widget.ProgressBar;

import com.example.vmmvexample1.adapter.RecyclerAdapter;
import com.example.vmmvexample1.model.NicePlace;
import com.example.vmmvexample1.viewmodels.MainActivityViewModel;
import com.google.android.material.floatingactionbutton.FloatingActionButton;

import java.util.List;

public class MainActivity extends AppCompatActivity {
	private static final String TAG = "MainActivity";

	private FloatingActionButton mFab;
	private RecyclerView mRecyclerView;
	private RecyclerAdapter mAdapter;
	private ProgressBar mProgressBar;
	private MainActivityViewModel mainActivityViewModel;


	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_main);
		//
		mFab = findViewById(R.id.fab);
		mRecyclerView = findViewById(R.id.recyclerView);
		mProgressBar = findViewById(R.id.progress_bar);
		mainActivityViewModel = ViewModelProviders.of(this).get(MainActivityViewModel.class);
		mainActivityViewModel.init();

		//
		mainActivityViewModel.getmListNicePlaces().observe(this, new Observer<List<NicePlace>>() {
			@Override
			public void onChanged(List<NicePlace> nicePlaces) {
				mAdapter.notifyDataSetChanged();
			}
		});

		mainActivityViewModel.getIsUpdating().observe(this, new Observer<Boolean>() {
			@Override
			public void onChanged(Boolean aBoolean) {
				if (aBoolean) {
					showProgressBar();
				} else {
					hideProgressBar();
					mRecyclerView.smoothScrollToPosition(mainActivityViewModel.getmListNicePlaces().getValue().size() -1);
				}

			}
		});

		mFab.setOnClickListener(v -> {
			mainActivityViewModel.addNewValue(new NicePlace("https://i.imgur.com/ZcLLrkY.jpg", "Washington"));
		});
		//
		initRecyclerView();
	}

	private void initRecyclerView() {
		mAdapter = new RecyclerAdapter(this, mainActivityViewModel.getmListNicePlaces().getValue());
		RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(this);
		mRecyclerView.setLayoutManager(layoutManager);
		mRecyclerView.setAdapter(mAdapter);
	}

	private void showProgressBar() {
		mProgressBar.setVisibility(View.VISIBLE);
	}

	private void hideProgressBar() {
		mProgressBar.setVisibility(View.GONE);
	}
}

File: AppLearnMatch/app/src/main/java/com/nghianv/applearnmatch/MainActivity.kt
package com.nghianv.applearnmatch

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    var number1: Int = 0
    var number2: Int = 0
    var ketqua: Int = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        button_cong.setOnClickListener {
            getNumberInput()
            ketqua = number1 + number2
            txt_ketqua.text = ketqua.toString()
        }
        button_tru.setOnClickListener {
            getNumberInput()
            ketqua = number1 - number2
            txt_ketqua.text = ketqua.toString()
        }
        button_nhan.setOnClickListener {
            getNumberInput()
            ketqua = number1 * number2
            txt_ketqua.text = ketqua.toString()
        }
        button_chia.setOnClickListener {
            getNumberInput()
            ketqua = number1 / number2
            txt_ketqua.text = ketqua.toString()
        }
    }

    fun getNumberInput() {
        number1 = (edt_number1.text.toString()).toInt()
        number2 = (edt_number2.text.toString()).toInt()
    }
}